message ProtocolHeader
{
    enum Command {
        INITIALIZE = 1;
        GET_ALL_MSG = 2;
        PUSH = 3;
        PUSH_ACK = 4;
    }

    required Command cmdid = 1;
    required int32 datalen = 2;
}
